pkg/scheduler/scheduler_test.go

```diff
@@ -63,3 +69,211 @@ func TestCreateScheduler(t *testing.T) {
+				Status:     workv1alpha2.ResourceBindingStatus{},
+			},
+			newScheduledCondition: successCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-1", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{successCondition}},
+			},
+		},
+		{
+			name: "add failure condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-2", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{},
+			},
+			newScheduledCondition: failureCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-2", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{failureCondition}},
+			},
+		},
+		{
+			name: "replace to success condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-3", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{failureCondition}},
+			},
+			newScheduledCondition: successCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-3", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{successCondition}},
+			},
+		},
+		{
+			name: "replace failure condition",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-4", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{successCondition}},
+			},
+			newScheduledCondition: failureCondition,
+			expected: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "rb-4", Namespace: "default"},
+				Spec:       workv1alpha2.ResourceBindingSpec{},
+				Status:     workv1alpha2.ResourceBindingStatus{Conditions: []metav1.Condition{failureCondition}},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := karmadaClient.WorkV1alpha2().ResourceBindings(test.binding.Namespace).Create(context.TODO(), test.binding, metav1.CreateOptions{})
+			if err != nil {
+				t.Error(err)
+			}
+			err = scheduler.patchBindingScheduleStatus(test.binding, test.newScheduledCondition)
+			if err != nil {
+				t.Error(err)
+			}
+			res, err := karmadaClient.WorkV1alpha2().ResourceBindings(test.binding.Namespace).Get(context.TODO(), test.binding.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Error(err)
+			}
+			if !reflect.DeepEqual(res, test.expected) {
+				t.Errorf("Test %s Not Equal", test.name)
```

Comment:
We don't need to log the test name as the name is already present in the result.
For example(I hacked the tests and let it fails) :
```
$ go test ./pkg/scheduler/... -run=Test_patchBindingScheduleStatus -v
=== RUN   Test_patchBindingScheduleStatus
=== RUN   Test_patchBindingScheduleStatus/add_success_condition
=== RUN   Test_patchBindingScheduleStatus/add_failure_condition
=== RUN   Test_patchBindingScheduleStatus/replace_to_success_condition
=== RUN   Test_patchBindingScheduleStatus/replace_failure_condition
    scheduler_test.go:171: Test replace failure condition Not Equal
```
The `Test_patchBindingScheduleStatus/replace_failure_condition` is exactly the test name.